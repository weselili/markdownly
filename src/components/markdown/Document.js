import React from 'react';
import Preview from './Preview';
import Editor from './Editor';
import styles from './Document.css';
import PropTypes from 'prop-types';

function Document({ document, handleChange }) {
  return (
    <>
      <h3>{document.title}</h3>
      <div className={styles.Document}>
        <Editor markdown={document.markdown} handleChange={handleChange} />
        <Preview markdown={document.markdown} />
      </div>
    </>
  );
}

Document.propTypes = {
  document: PropTypes.object.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Document;
