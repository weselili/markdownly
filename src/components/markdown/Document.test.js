import React from 'react';
import { shallow } from 'enzyme';
import Document from './Document';

describe('Document component', () => {
  it('renders Document', () => {
    const wrapper = shallow(<Document />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
