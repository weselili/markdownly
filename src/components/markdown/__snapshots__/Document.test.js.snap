// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document component renders Document 1`] = `
<Fragment>
  <div
    className="Document"
  >
    <Editor
      handleChange={[Function]}
      markdown=""
    />
    <Preview
      markdown=""
    />
  </div>
</Fragment>
`;
